process/block/baseProcess.go

```diff
@@ -2110,3 +2115,19 @@ func (bp *baseProcessor) setNonceOfFirstCommittedBlock(nonce uint64) {
 	bp.nonceOfFirstCommittedBlock.HasValue = true
 	bp.nonceOfFirstCommittedBlock.Value = nonce
 }
+
+func (bp *baseProcessor) checkSentSignaturesBeforeCommitting(header data.HeaderHandler) error {
+	validatorsGroup, err := headerCheck.ComputeConsensusGroup(header, bp.nodesCoordinator)
+	if err != nil {
+		return err
+	}
+
+	validatorsPKs := make([][]byte, 0, len(validatorsGroup))
+	for _, validator := range validatorsGroup {
+		validatorsPKs = append(validatorsPKs, validator.PubKey())
+	}
+
+	bp.sentSignaturesTracker.ResetCountersManagedBlockSigners(validatorsPKs)
```

Comment:
perhaps `ResetCountersManagedBlockSigner`, thus no need for second iteration on `sentSignaturesTracker` component?

Reply:
interesting. Yes, I think we can refactor this to gain performance. Thanks